src/library/scala/collection/StringParsers.scala

```diff
@@ -0,0 +1,283 @@
+package scala
+package collection
+
+import scala.annotation.tailrec
+
+/** A module containing the implementations of parsers from strings to numeric types, and boolean
+ */
+final private[scala] object StringParsers {
+
+  //compile-time constant helpers
+
+  //Int.MinValue == -2147483648
+  private final val intOverflowBoundary = -214748364
+  private final val intOverflowDigit = 9
+  //Long.MinValue == -9223372036854775808L
+  private final val longOverflowBoundary = -922337203685477580L
+  private final val longOverflowDigit = 9
+
+  
+  @inline
+  private def decValue(ch: Char): Int = java.lang.Character.digit(ch, 10)
+
+  //bool
+  @inline
+  def parseBool(from: String): Option[Boolean] = from.toLowerCase match {
+    case "true" => Some(true)
+    case "false" => Some(false)
+    case _ => None
+  }
+
+  //integral types
+  @inline
```

Comment:
Why is this inline?  Seems like enough code that it should not be inlined in most cases.